tests/ensemble_evaluator/test_ensemble_evaluator.py

```diff
@@ -190,8 +191,10 @@ def test_dispatchers_can_connect_and_monitor_can_shut_down_evaluator(evaluator):
 
 
 def test_monitor_stop(evaluator):
-    asyncio.set_event_loop(asyncio.new_event_loop())
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
     monitor = evaluator.run()
     events = monitor.track()
     snapshot = Snapshot(next(events).data)
     assert snapshot.get_status() == "Unknown"
+    loop.close()
```

Comment:
So this is needed? 
1) Doesn't this happen when `loop` goes out of scope?
2) If yes to 1), then does that mean that something is running after the test completes?

Reply:
A typical event loop runs as long as the application if we don't call close() on it the OS will clean up the resources on program exit

Calling loop.close() is important for event loops that have a clear lifetime. For example, a library might create a fresh event loop for a specific task, run it in a dedicated thread, and dispose of it. Failing to close such a loop could leak its internal resources (such as the pipe it uses for inter-thread wakeup) and cause the program to fail. Another example are test suites, which often start a new event loop for each unit test to ensure separation of test environments.

Found this information in the following stackoverflow answer:
https://stackoverflow.com/questions/50279522/when-is-the-right-time-to-call-loop-close#:~:text=Since%20a%20typical%20event%20loop,that%20have%20a%20clear%20lifetime.

Not really sure why an event loop is needed in this test anyway.